### ISSUE TITLE
Jump to definition for RIot.js components is not working

### ISSUE THREAD

**Issue by GianlucaGuarini** [Kal-Aster/vscode-riotjs#13]
I am not sure if this issue is related to https://github.com/Kal-Aster/vscode-riotjs/issues/9 but the components jump to definition doesn't work. For example:

```riot
<my-component>
  <child/> <!-- I expect the jump to definition here to jump to child.riot -->

  <script>
    // jump to definition here doesn't work here either
    import Child from './child.riot'

    export default {
      components: { Child }
    }
  </script>
</my-component>
```

**Comment by Kal-Aster** [comment]
`<!-- I expect the jump to definition here to jump to child.riot -->`
This one isn't actually implemented.

`    // jump to definition here doesn't work here either`
This one should work, I'll check it out as soon as possible.

**Comment by Kal-Aster** [comment]
The extension indeed don't work sometimes and I'm trying to find the issue, but in my setup it wasn't consistent.

Did the jump to definition feature never worked for you?
Is the code you shared that you've tested the extension on?

**Comment by GianlucaGuarini** [comment]
> Did the jump to definition feature never worked for you?

Yes it did work in intellij https://plugins.jetbrains.com/plugin/12748-riot-js

> Is the code you shared that you've tested the extension on?

Yes basically you can take any of the components inside the riot/test folder from the official riot repo
Please check this video, the jump to definition works only within the script tag to the import definition. If I want to jump to the imported file from the `import` statement or from the html it doesn't work

https://github.com/user-attachments/assets/7b3885b6-c1b6-415b-8e64-eb9a15455a49

**Comment by Kal-Aster** [comment]
I see... That's actually not my extension 😅 [This one is](https://marketplace.visualstudio.com/items?itemName=KalAster.vscode-riotjs).
I made it only for VSCode, for now.

However I tested working in some projects of mine and I noticed that after some time the extension breaks, especially when opening or closing many times multiple riot documents... So for sure there are many optimisations to do, but meanwhile if you can try it on VSCode.

**Comment by Kal-Aster** [comment]
> However I tested working in some projects of mine and I noticed that after some time the extension breaks, especially when opening or closing many times multiple riot documents... So for sure there are many optimisations to do, but meanwhile if you can try it on VSCode.

Actually the problem was in the autoclose feature. I fixed it and now it should work consistently.

**Comment by GianlucaGuarini** [comment]
I have tested it in VsCode as well and it doesn't seem to work:

![Screenshot 2025-01-03 at 23 30 04](https://github.com/user-attachments/assets/efd8291d-c315-4161-9bdf-03eab0c94c98)

**Comment by Kal-Aster** [comment]
Really sorry for the late response...

I'm actually using it and besides many performance issues I get the jump to definition feature.
Maybe you're not getting it because there are some other errors or perhaps it's due to the performance issues.

Can you help me understand better the issue? 
Having the same code you're testing it to would be helpful.

**Comment by GianlucaGuarini** [comment]
Sure besides the Typo 😄  that's the file https://github.com/riot/riot/blob/main/test/components/dashed-attribute-parent.riot

**Comment by Kal-Aster** [comment]
The typo shouldn't be a problem ahah

Anyway I opened it and it doesn't work indeed. I'm on it.

**Comment by Kal-Aster** [comment]
The issue was that the jump to definition feature expected the riot document to have a script defined that _should_ be used to map the original definition. The mapping is still to do, but meanwhile the jump to definition should work properly, now
